hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/PerformanceEvaluation.java

```diff
@@ -1998,6 +2006,41 @@ protected void testTakedown() throws IOException {
       super.testTakedown();
     }
   }
+  static class MetaRandomReadTest extends TableTest {
+    private Random rd = new Random();
+    private RegionLocator regionLocator;
+
+    MetaRandomReadTest(Connection con, TestOptions options, Status status) {
+      super(con, options, status);
+      LOG.info("call getRegionLocation");
+    }
+
+    @Override
+    void onStartup() throws IOException {
+      this.table = connection.getTable(TableName.valueOf("hbase:meta"));
+      this.regionLocator = connection.getRegionLocator(TableName.valueOf("hbase:meta"));
+    }
+
+    @Override
+    boolean testRow(final int i, final long startTime) throws IOException, InterruptedException {
+      if (opts.randomSleep > 0) {
+        Thread.sleep(rd.nextInt(opts.randomSleep));
+      }
+      HRegionLocation hRegionLocation = regionLocator.getRegionLocation(Bytes.toBytes(Integer.toString(rd.nextInt(100) + 1)), true);
```

Comment:
the return value is not used?

Reply:
Right.

Reply:
MetaWriteTest should be executed before MetaReadTest to populate the table.

Reply:
Can we generate a correct key, let's say 'aaabc' and get the location, and make sure that the key falls in the boundary of [region.startKey, region.endKey), it will be great.

Reply:
For PE meta read testing, at the end of the test, can we get the read/write count for all meta regions? This will help us to see if read load is distributed among all meta replica regions.

Reply:
> Right. I was thinking to add trace in the case of troubleshooting but not sure.

Then lt's remove it? Otherwise there will be a checkstyle warning.

Reply:
added logging

Reply:
Added split key generation so write, read and clean up will work with regions with continuous range of start key and end key.